tests/ui/must_use_candidates.rs

```diff
@@ -84,7 +84,7 @@ pub unsafe fn mutates_static() -> usize {
 }
 
 #[no_mangle]
-pub fn unmangled(i: bool) -> bool {
+pub extern "C" fn unmangled(i: bool) -> bool {
```

Comment:
Just applying this lint. Helps us avoid adding noise to `tests/ui/must_use_candidates.fixed`